### ISSUE TITLE
why get the same index_array by shuffling in keras.engine.training._fit_loop

### ISSUE THREAD

**Issue by dijiuxinsheng** [fchollet/keras#4950]
i have set np.random.seed(xx) at the begining, then i check the index_array of each epoch, and i found the index_array of each (different) epoch is the same.
anyone knows why

code in  keras.engine.training._fit_loop
`for epoch in range(initial_epoch, nb_epoch):
            callbacks.on_epoch_begin(epoch)
            if shuffle == 'batch':
                index_array = batch_shuffle(index_array, batch_size)
            elif shuffle:
                np.random.shuffle(index_array)`
